core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -246,18 +246,18 @@ private[spark] class MemoryStore(
         val amountToRequest = size - unrollMemoryUsedByThisBlock
         keepUnrolling = reserveUnrollMemoryForThisTask(blockId, amountToRequest, memoryMode)
         if (keepUnrolling) {
-          unrollMemoryUsedByThisBlock += amountToRequest
+          unrollMemoryUsedByThisBlock = size
         }
+      } else if (size < unrollMemoryUsedByThisBlock) {
```

Comment:
#19285 miss this check before? Oops! This could waste storage memory. Thanks for adding this.
Also, cc @cloud-fan @ConeyLiu

Reply:
In #19285, we first release `unrollMemoryUsedByThisBlock` unroll memory, and then we request `entry.size` storage memory. So, there is no waste of resources here.

Reply:
There is no problem before,because it releases all unroll memory in 257 line

Reply:
Ah, I see, glad to hear that.